sdk/typescript/src/types/transactions.ts

```diff
@@ -107,6 +107,39 @@ export type OwnedObjectRef = {
   reference: SuiObjectRef;
 };
 
+export type DevInspectResults = {
+  effects: TransactionEffects;
+  results: DevInspectResultsType;
+};
+
+export type ResultType = 'Ok' | 'Err;';
+export type DevInspectResultsType = {
+  result: ResultType;
+  contents?: DevInspectResultTupleType[];
+  error?: string;
+};
+
+export type DevInspectResultTupleType = {
+  index: number;
+  executionResult: ExecutionResultType;
+};
+
+export type ExecutionResultType = {
+  mutableReferenceOutputs: MutableReferenceOutputType[];
+  returnValues: ReturnValueType[];
+};
+
+export type MutableReferenceOutputType = {
```

Comment:
@666lcz This is a tuple in Rust, I saw codes around using structs like this to handle tuple in Rust as well, so presumably this should work?

Reply:
This won't work. It will just deserialize to a list. Can we change the type of `mutable_reference_outputs` on the rust side to a struct instead of tuple?